pkg/trie/trie.go

```diff
@@ -13,17 +13,23 @@ import (
 // EmptyHash is the empty trie hash.
 var EmptyHash = common.MustBlake2bHash([]byte{0})
 
-type ChildTrieSupport interface {
+type ReadChildTries interface {
```

Comment:
Nit/maybe can model like below with `PrefixTrieRead` and change to `ChildTriesRead` and `ChildTriesWrite`. This is just an idea, do whatever you think works best here